common/common.cpp

```diff
@@ -2516,6 +2527,37 @@ bool create_directory_with_parents(const std::string & path) {
 #endif // _WIN32
 }
 
+std::string get_cache_directory() {
```

Comment:
Because this function also create, not just simple get (because you called `create_directory_with_parents` inside), so it's better to rename it to `create_cache_directory()`

Reply:
That's a good point. Since the function doesn't always create the directory, I think these two alternatives might be better:

1. rename the function to `get_or_create_cache_directory()`
2. keep the current name and move the directory creation logic out of the function:
```cpp
std::string cache_directory = get_cache_directory();
const bool success = create_directory_with_parents(cache_directory);
...
```

What do you think?

Reply:
IMO the 2nd proposal makes more sense to me. We should prevent having "one function does all". Breaking it into small steps make these functions easier to be re-used later on.